pkg/config/config.go

```diff
@@ -39,16 +47,30 @@ func FromYAMLFile[T any, P interface {
 	return c, nil
 }
 
-// ParseFlags parses CLI flags and
-// returns a new value of type T created from them.
-func ParseFlags[T any]() (*T, error) {
-	var f T
+// ParseFlags parses CLI flags and stores the result
+// in the value pointed to by v. If v is nil or not a pointer,
+// ParseFlags returns an [ErrInvalidArgument] error.
+// ParseFlags adds a default Help Options group,
+// which contains the options -h and --help.
+// If either option is specified on the command line,
+// ParseFlags prints the help message to [os.Stdout] and exits.
+func ParseFlags(v any) error {
+	rv := reflect.ValueOf(v)
+	if rv.Kind() != reflect.Pointer || rv.IsNil() {
+		return errors.Wrapf(ErrInvalidArgument, "non-nil pointer expected, got %T", v)
+	}
 
-	parser := flags.NewParser(&f, flags.Default)
+	parser := flags.NewParser(v, flags.Default^flags.PrintErrors)
```

Comment:
It would be nice if you could provide some comments why you are clearing the `flags.PrintErrors` bits. If I understand it correctly, you are doing this so that the help message is not printed twice since you are explicitly printing it. However, wouldn't it be sufficient to rely on the library and just exit down below if the error is of type `flags.ErrHelp` without having to clear that flag?

Reply:
The function returns either `ErrInvalidArgument` or a flag-related error. In the latter case, the library would print the error and the function would then return this error, which we would then print again. Therefore, the actual error handling is the responsibility of the caller. I can mention this briefly in the documentation.